lib/LaTeXML/Package/utf8.def.ltxml

```diff
@@ -29,6 +29,17 @@ AssignValue(INPUT_ENCODING => undef);
 # And set the Mouth to simply pass-thru the utf8.
 AssignValue(PERL_INPUT_ENCODING => 'UTF8');
 
+DefPrimitive('\DeclareUnicodeCharacter Digested {}', sub {
```

Comment:
Looks quite plausible, but wouldn't Expanded be more appropriate than Digested?  A simple example of use would be helpful (if not an actual test case).

Reply:
Very good point, I don't think there is any use case for Digested. It's just me picking the wrong option (still learning!). After reading the `inputenc` source, I believe Expanded is the good answer: the code responsible for reading the codepoint is `\count@"#1\relax`, which I think expands `#1` without executing.

But you made notice something else: the current code mangles spaces incorrectly. `inputenc` wraps the second argument in the macro `\IeC{}`, to prevent active characters from gobbling spaces. Plus `\IeC` acts as `\protect`, and that is also important to reproduce.

It's likely that `\IeC` is needed in `\DeclareInputText` too. I'll come up with some test cases and think about a sensible implementation.

Reply:
I think I understand now. The argument should definitely be Expanded. If you pass unexpandable tokens, they will get executed in the middle of `utf8.def`, and that is definitely not the intended use of `inputenc`.

Regarding the `\IeC{}` trick in `inputenc`, it is used to "avoid any space after the command being gobbled up when the text is written out to an auxiliary file" (from `utf8ienc.dtx`). LaTeXML does not write auxiliary files, and I gather it does not emulate writing to them, so the `\IeC{}` trick is not needed. Likewise for the `\protect` hidden inside `\IeC{}`. So if I am correct on this, the current implementation is fine.

Do you need a `t/encoding/utf8.tex`? I can write some examples (e.g. mapping € to `\euro`, or √ to `\sqrt`), even though I can't think of any edge case that needs to be tested.